content/blog/understanding-event-loop/index.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Understanding event loop in JavaScript"
+date: "2020-09-16"
+coverImage: "index.png"
+author: "Aman Agrawal"
+tags: ["JavaScript", "event-loop", "call stack", "event queue", "multi-threaded"]
+description: "Learn the basic concepts about the JavaScript Event Loop."
+---
+
+We often heard that JavaScript is a single-threaded programming language, which means it executes all of the instructions line by line in a synchronous manner. Thus since everything works on the main thread, there seems to be no possibility of executing parallel processes in JavaScript.
+
+But now let's assume that we have a function which is taking some time to execute, for example in the below function which is having a loop over 10k times, the `console.log()` would be executed once the loop is over and in the meantime, our UI interaction with the browser would be interrupted.
+
+```JavaScript
+function someTimeTakingFunc() {
+  for(var i = 0; i < 10; i++) {
+
+  }
+  console.log("Loop has been executed")
+}
+```
+
+The above scenario seems to be impractical, but there might be cases where any function can take an indeterminate amount of time and our main thread would be blocked, for example calling an API for fetching data from the server-side. And this is the practical use case we are dealing with in development every day. So how JavaScript handles this and where does the event loop comes into the picture? We will surely get to know this, but before moving further, let's understand the basic memory architecture in JavaScript.
+
+### Memory Organization of JavaScript: 
+
+The Javascript Engine consists of two main components:
+ -  **Memory Heap** — this is where the memory allocation happens, all of our object variables are assigned here in a random manner.
+ -  **Call Stack** — this is where your function calls are stored.
+
+ ### Understanding the Call Stack
+
+Call stack is a LIFO (last in first out) data structure. All of the function calls are pushed into this call stack and are said to be a frame. In short function, a stack is nothing but the simple stack data structure which keeps tracks of the function currently being executed.
```

Comment:
` which keeps tracks of the function` should be ` which keeps track of the function`